### ISSUE TITLE
Include `{data.table}` example with and without lint for `redundant_equals_linter()` docs

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2657]
Since the lint message highlights this case so prominently, there should also be a corresponding example in the docs:

``` r
library(lintr)

lint(text = "dt[x == FALSE, y]", linters = redundant_equals_linter())
#> <text>:1:4: warning: [redundant_equals_linter] Using == on a logical vector is redundant. Well-named logical vectors can be used directly in filtering. For data.table's `i` argument, wrap the column name in (), like `DT[(is_treatment)]`.
#> dt[x == FALSE, y]
#>    ^~~~~~~~~~

lint(text = "dt[!(x), y]", linters = redundant_equals_linter())
#> ℹ No lints found.
```

<sup>Created on 2024-09-06 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
